src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -68,10 +74,22 @@ public Artefact createPublication(Artefact artefact, MultipartFile file) {
             file
         );
         artefact.setPayload(blobUrl);
-
         return artefactRepository.save(artefact);
     }
 
+
+    /**
+     * Checks if the artefact has a display from date of today or previous then triggers the sub fulfilment
+     * process on subscription-management if appropriate.
+     */
+    public List<Subscription> checkAndTriggerSubscriptionManagement(Artefact artefact) {
+        if (artefact.getDisplayFrom().isBefore(LocalDateTime.now().plusDays(1))) {
+            return subscriptionManagementService.sendSubTrigger(artefact);
```

Comment:
I think the logic for the date checking will need updating a bit, as time has the chance to be included as well in dates so not sure the plus one days will work.

Also I would check the displayTo instead, just to make sure it's in the range.

Reply:
Would be good if you could take another look at this. Seems to work in all cases, including that of a general_publication with no date from/to published